### PR TITLE
Implement a benchmark algorithm `Delaunay2D`

### PR DESCRIPTION
**Pull request by Chubercik**
- Implements 🟪**Algorithm**🟪 **_Delaunay2D_**: `Delaunay 1000 points in 2D` benchmark of the **Math** category proposed in #36.

### CODE REVIEW

**Comment by reduz** [comment]
Wow, great work!

File: benchmarks/math/delaunay2D.gd [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+			rng.randf_range(upper_left.x, bottom_right.x),
+			rng.randf_range(upper_left.y, bottom_right.y)
+		))
+
+	return points
+
+var scattered_points := scatter_points(NUM_POINTS, UPPER_LEFT, BOTTOM_RIGHT)
+
+func _init() -> void:
+	test_idle = true
```
Comment (Calinou) [inline benchmarks/math/delaunay2D.gd:26]: Like in https://github.com/godotengine/godot-benchmarks/pull/37, this isn't effective for the same reasons.

**Review by Calinou — CHANGES_REQUESTED** [review]
Tested locally, it works as expected.

However, this benchmark runs in roughly 2 ms on my i9-13900K with an optimized Linux editor binary. This means there's high potential for variance due to differences in setup time. We want to make sure we are benchmarking the actual triangulation logic, not the time it takes for the script to be parsed and compiled.

I suggest making it run more iterations so that running the benchmark requires 50× more time (it should take at least 100 ms on my CPU).

PS: `delaunay2D.gd` should be renamed to `delaunay_2d.gd` for consistency with other file names in the project.

**Comment by Chubercik** [comment]
Committed an update which changes the number of points used by the algorithm from `1000` to `7000`, which on my setup ups the benchmark's execution time from `~5.5ms` to `~250ms` - if this time I overdid the time it takes to complete be sure to hit me up and I'll fiddle with it further.

**Review by Calinou — APPROVED** [review]
It now takes 75 ms to run on my setup with a release build, which should be good enough for accurate measurements.

**Pull request merged**